src/app/clusters/network-commissioning/NetworkCommissioningLogic.h

```diff
@@ -0,0 +1,242 @@
+                    DeviceLayer::NetworkCommissioning::WiFiScanResponseIterator * networks) override;
+
+    // ThreadDriver::ScanCallback
+    void OnFinished(DeviceLayer::NetworkCommissioning::Status err, CharSpan debugText,
+                    DeviceLayer::NetworkCommissioning::ThreadScanResponseIterator * networks) override;
+
+    // Actual handlers of the commands
+    std::optional<DataModel::ActionReturnStatus>
+    HandleScanNetworks(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                       const NetworkCommissioning::Commands::ScanNetworks::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus>
+    HandleAddOrUpdateWiFiNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                                 const NetworkCommissioning::Commands::AddOrUpdateWiFiNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus>
+    HandleAddOrUpdateThreadNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                                   const NetworkCommissioning::Commands::AddOrUpdateThreadNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus>
+    HandleRemoveNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                        const NetworkCommissioning::Commands::RemoveNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus>
+    HandleConnectNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                         const NetworkCommissioning::Commands::ConnectNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus>
+    HandleReorderNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                         const NetworkCommissioning::Commands::ReorderNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus> HandleNonConcurrentConnectNetwork();
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+    std::optional<DataModel::ActionReturnStatus>
+    HandleQueryIdentity(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                        const NetworkCommissioning::Commands::QueryIdentity::DecodableType & req);
+
+    std::optional<DataModel::ActionReturnStatus>
+    HandleAddOrUpdateWiFiNetworkWithPDC(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                                        const NetworkCommissioning::Commands::AddOrUpdateWiFiNetwork::DecodableType & req);
+#endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+
+    // Attribute handling
+    CHIP_ERROR SetInterfaceEnabled(bool enabled) { return mpBaseDriver->SetEnabled(enabled); }
+    uint8_t GetMaxNetworks() const { return mpBaseDriver->GetMaxNetworks(); }
+    uint8_t GetScanNetworkTimeoutSeconds() const
+    {
+        return mpWirelessDriver != nullptr ? mpWirelessDriver->GetScanNetworkTimeoutSeconds() : 0;
```

Comment:
Realistically this should only be called when we have a wireless driver, right?  It's weird to be returning a valid-looking value when we are in a known-bad state...  Not sure the codesize cost of not doing it is worth it, though, but it might be worth documenting what situations here are expected vs not expected.